src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -84,4 +97,13 @@ public boolean equals(Object other) {
         return id.getText().equals(card.id.getText())
                 && person.equals(card.person);
     }
+
+    private static String getColorForTag(String tagValue) {
+        if (!tagColors.containsKey(tagValue)) {
+            tagColors.put(tagValue, UiStyle.getInstance().getRandomHexColor());
+        }
+
+        return tagColors.get(tagValue);
+    }
```

Comment:
Method name is misleading because the method works as both a getter and a setter.

Reply:
What would be a more appropriate name for the method? Do I split the "If" part into another method?